tests/test_release_transpiler.py

```diff
@@ -32,13 +37,83 @@ def test_build_release_matrix(tmpdir):
+)
+def test_build_release_matrix_none_py_coords(
+    tmpdir, py_coords_input, packages_lib2, py_coords_not_in_lib3
+):
+    """lib1 tests packages with builtins,
+    lib2 tests packages with same version for two py coordinates,
+    lib3 tests packages with different versions for each py coordinate"""
+    release_base = "2020.01.a1"
+    release_folder = os.path.join(_get_test_root(), "data/test_releases/")
+    with tmpdir.as_cwd():
+        build_matrix_file(release_base, release_folder, builtins, py_coords_input)
+        new_release_file = "{}.yml".format(release_base)
+        assert os.path.isfile(new_release_file)
+        with open(new_release_file) as f:
+            release_matrix = yaml.safe_load(f)
+
+        assert release_matrix["lib1"] == builtins["lib1"]
+        assert release_matrix["lib2"] == packages_lib2
+        assert all(
+            py_coordinate not in list(release_matrix["lib3"].keys())
+            for py_coordinate in py_coords_not_in_lib3
+        )
+
+
+@pytest.mark.parametrize(
+    "matrix",
+    [({"py": ["3.8"], "rhel": ["7"]}), ({"py": ["3.8", "3.10"], "rhel": ["7", "8"]})],
+)
+def test_transpile_add_argument(tmpdir, matrix):
     release_file = os.path.join(_get_test_root(), "data", "test_release_matrix.yml")
     release_base = os.path.basename(release_file).strip(".yml")
     with tmpdir.as_cwd():
-        transpile_releases(release_file, os.getcwd())
-        for rhel_ver in ("rhel7",):
-            for py_ver in ("py38",):
+        transpile_releases(release_file, os.getcwd(), matrix)
+        for rhel_coordinate in matrix["rhel"]:
+            rhel_coordinate_filename_format = f"rhel{rhel_coordinate}"
+            for py_coordinate in matrix["py"]:
+                py_coordinate_filename_format = f"py{py_coordinate.replace('.', '')}"
                 assert os.path.isfile(
-                    "{}-{}-{}.yml".format(release_base, py_ver, rhel_ver)
+                    "{}-{}-{}.yml".format(
+                        release_base,
+                        py_coordinate_filename_format,
+                        rhel_coordinate_filename_format,
+                    )
                 )
+
+
+@pytest.mark.parametrize(
+    "matrix,reason_for_failure",
+    [
+        ({"py": ["3.8"], "rhel": ["7"]}, "Test passes, no error reported"),
+        (
+            {"py": ["3.7"], "rhel": ["7"]},
+            "Python version py37 not found for rhel version rhel7 for package lib1.",
+        ),
+        (
+            {"py": ["3.6"], "rhel": ["5"]},
+            "Rhel version rhel5 not found for package lib1.",
+        ),
+    ],
+    ids=["Pass for all packages", "Fail", "Fail"],
+)
+def test_check_version_exists_for_coordinates(matrix, reason_for_failure):
+    release_file = os.path.join(_get_test_root(), "data", "test_release_matrix.yml")
+    try:
+        transpile_releases(release_file, os.getcwd(), matrix)
+    except KeyError as exception_info:
+        assert reason_for_failure in str(exception_info)
```

Comment:
If I stop the test at this point and `print(str(exception_info))`, I get:
`"'Rhel version rhel5 not found for package lib1.'"`
I don't understand why it has `'"` and how to remove `'`.
Do you have an idea for that?

Reply:
oh ehm hmmm not sure - it's the quotation marks around the whole thing that you want to get rid of? how do you get the output / print ? do you use a debugger? is the test failing because of the quotation marks?

in general, i think you could match for a smaller part of the error message, to make the test less brittle when somebody should decide to change the error message - one could argue that we don't really care about the wording, but rather that some key information is present, like

```python
except KeyError as exception_info:
    for info in expected_info:
        assert info in str(exception_info)
```

where `expected_info` would for instance be `["py37", "rhel7","lib1"]` for the second test (and you could still keep the full text reason as the test id, for better understanding of what the test is testing / why it should fail)

Reply:
Nice comment, I use the pattern all(), but have same effect.

Reply:
great!